### ISSUE TITLE
shake 0.15 breaks Hoogle 4.2.36

### ISSUE THREAD

**Issue by deepfire** [ndmitchell/hoogle#106]
```
[72 of 75] Compiling Recipe.Command   ( src/Recipe/Command.hs, dist/build/hoogle/hoogle-tmp/Recipe/Command.dyn_o )

src/Recipe/Command.hs:12:9:
    Ambiguous occurrence ‘CmdLine’
    It could refer to either ‘Development.Shake.CmdLine’,
                             imported from ‘Development.Shake’ at src/Recipe/Command.hs:7:1-24
                             (and originally defined in ‘Development.Shake.Command’)
                          or ‘CmdLine.All.CmdLine’,
                             imported from ‘CmdLine.All’ at src/Recipe/Command.hs:9:1-18
                             (and originally defined in ‘CmdLine.Type’
                                at src/CmdLine/Type.hs:(19,1)-(57,35))
builder for `/nix/store/rhm3bxi2khkrjq6762vs4vn58fcbxaar-haskell-hoogle-ghc7.8.3-4.2.36-shared.drv' failed with exit code 1
```

**Comment by ndmitchell** [comment]
Whoops. Fixing now.

**Comment by ndmitchell** [comment]
All fixed, along with all the warnings that cropped up. Released as 4.2.39.

**Comment by deepfire** [comment]
Thank you!

**Comment by OlegTsyba** [comment]
shake-0.15.1 break hoogle-4.2.40 again

    Configuring shake-0.15.1...
    Building shake-0.15.1...
    Failed to install shake-0.15.1
    Build log ( /home/oleh/.cabal/logs/shake-0.15.1.log ):
    [1 of 1] Compiling Main             ( /tmp/shake-0.15.1-8397/shake-0.15.1/dist/setup/setup.hs, /tmp/shake-0.15.1-8397/shake-0.15.1/dist/setup/Main.o )
    Linking /tmp/shake-0.15.1-8397/shake-0.15.1/dist/setup/setup ...
    Configuring shake-0.15.1...
    Building shake-0.15.1...
    Preprocessing library shake-0.15.1...
    [ 1 of 47] Compiling General.Bilist   ( src/General/Bilist.hs, dist/build/General/Bilist.o )
    [ 2 of 47] Compiling General.Template ( src/General/Template.hs, dist/build/General/Template.o )
    [ 3 of 47] Compiling General.Binary   ( src/General/Binary.hs, dist/build/General/Binary.o )
    [ 4 of 47] Compiling General.Cleanup  ( src/General/Cleanup.hs, dist/build/General/Cleanup.o )
    [ 5 of 47] Compiling General.Concurrent ( src/General/Concurrent.hs, dist/build/General/Concurrent.o )
    [ 6 of 47] Compiling General.Extra    ( src/General/Extra.hs, dist/build/General/Extra.o )
    [ 7 of 47] Compiling Development.Shake.Monad ( src/Development/Shake/Monad.hs, dist/build/Development/Shake/Monad.o )
    [ 8 of 47] Compiling General.Timing   ( src/General/Timing.hs, dist/build/General/Timing.o )
    [ 9 of 47] Compiling Development.Shake.Pool ( src/Development/Shake/Pool.hs, dist/build/Development/Shake/Pool.o )
    [10 of 47] Compiling Development.Shake.Resource ( src/Development/Shake/Resource.hs, dist/build/Development/Shake/Resource.o )
    [11 of 47] Compiling Paths_shake      ( dist/build/autogen/Paths_shake.hs, dist/build/Paths_shake.o )
    [12 of 47] Compiling Development.Shake.Profile ( src/Development/Shake/Profile.hs, dist/build/Development/Shake/Profile.o )
    [13 of 47] Compiling Development.Shake.ByteString ( src/Development/Shake/ByteString.hs, dist/build/Development/Shake/ByteString.o )
    [14 of 47] Compiling Development.Ninja.Env ( src/Development/Ninja/Env.hs, dist/build/Development/Ninja/Env.o )
    [15 of 47] Compiling Development.Ninja.Type ( src/Development/Ninja/Type.hs, dist/build/Development/Ninja/Type.o )
    [16 of 47] Compiling Development.Ninja.Lexer ( src/Development/Ninja/Lexer.hs, dist/build/Development/Ninja/Lexer.o )
    [17 of 47] Compiling Development.Ninja.Parse ( src/Development/Ninja/Parse.hs, dist/build/Development/Ninja/Parse.o )
    [18 of 47] Compiling Development.Shake.FilePath ( src/Development/Shake/FilePath.hs, dist/build/Development/Shake/FilePath.o )
    
    src/Development/Shake/FilePath.hs:11:5:
        Ambiguous occurrence `-<.>'
        It could refer to either `Development.Shake.FilePath.-<.>',
                                 defined at src/Development/Shake/FilePath.hs:105:1
                              or `System.FilePath.-<.>',
                                 imported from `System.FilePath' at src/Development/Shake/FilePath.hs:(19,1)-(25,5)
                                 (and originally defined in `System.FilePath.Posix')
    
    src/Development/Shake/FilePath.hs:11:5:
        Conflicting exports for `-<.>':
           `module System.FilePath' exports `System.FilePath.-<.>'
             imported from `System.FilePath' at src/Development/Shake/FilePath.hs:(19,1)-(25,5)
             (and originally defined in `System.FilePath.Posix')
           `(-<.>)' exports `Development.Shake.FilePath.-<.>'
             defined at src/Development/Shake/FilePath.hs:105:1
    cabal: Error: some packages failed to install:
    hoogle-4.2.40 depends on shake-0.15.1 which failed to install.
    shake-0.15.1 failed during the building phase. The exception was:
    ExitFailure 1

**Comment by ndmitchell** [comment]
@olegtsyba, what is your ghc version and filepath version? Use ghc --version and ghc-pkg list

**Comment by OlegTsyba** [comment]
The Glorious Glasgow Haskell Compilation System, version 7.6.3
filepath-1.4.0.0

**Comment by ndmitchell** [comment]
You are using an old GHC with a new filepath, which is causing the break in Shake. I'll put out a new release which fixes that in the next few days. Upgrading to GHC 7.10 would also fix it, or downgrading to an older filepath (the one that came installed with that GHC works).

**Comment by OlegTsyba** [comment]
Thanks!

**Comment by ndmitchell** [comment]
I've just released shake-0.15.2 which I believe should fix this issue.
